modules/core/src/main/java/org/apache/synapse/aspects/flow/statistics/tracing/opentelemetry/management/handling/span/SpanHandler.java

```diff
@@ -154,22 +209,21 @@ private void startSpan(StatisticDataUnit statisticDataUnit, MessageContext synCt
         // We only need to extract span context from headers when there are trp headers available
         if (isOuterLevelSpan(statisticDataUnit, spanStore) && headersMap != null) {
             // Extract span context from headers
-            spanContext = tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMapAdapter(headersMap));
-            span = tracer.buildSpan(statisticDataUnit.getComponentName()).asChildOf(spanContext).start();
+            Context context = extract(headersMap);
+            span = tracer.spanBuilder(statisticDataUnit.getComponentName()).setParent(context).startSpan();
         } else {
-            span = tracer.buildSpan(statisticDataUnit.getComponentName()).asChildOf(parentSpan).start();
-            spanContext = span.context();
+            span = tracer.spanBuilder(statisticDataUnit.getComponentName()).setParent(Context.current()
+                    .with(parentSpan)).startSpan();
         }
```

Comment:
Can we optimize this as the follows? Or do `extract(headersMap)` and `Context.current().with(parentSpan)` have separate return types?
```java
        Context context;
        if (isOuterLevelSpan(statisticDataUnit, spanStore) && headersMap != null) {
            // Extract span context from headers
            context = extract(headersMap);
        } else {
            context = Context.current().with(parentSpan);
            
        }
        span = tracer.spanBuilder(statisticDataUnit.getComponentName()).setParent(context).startSpan();
```

Reply:
Yes we can and it returns the same type. Resolved after a discussion.